tidy3d/components/simulation.py

```diff
@@ -437,6 +437,48 @@ def _warn_monitor_simulation_frequency_range(cls, val, values):
                 )
         return val
 
+    @pydantic.validator("monitors", always=True)
+    def diffraction_efficiency_size(cls, val, values):
+        """If any :class:`DiffractionEfficiencyMonitor` exists, ensure that its size is at least
+        as large as the simulation domain in the transverse direction."""
+        monitors = val
+        sim_size = values.get("size")
+        for monitor in monitors:
+            if not isinstance(monitor, DiffractionEfficiencyMonitor):
+                continue
+            _, (n_x, n_y) = monitor.pop_axis([0, 1, 2], axis=monitor.axis)
```

Comment:
I think this check doesn't take the `monitor.center` or `simulation.center` into account? For example, there could be a situation where the monitor is larger than the sim, but because of differing center locations, does not encapsulate it.